File: python-reprojector/pyreproj-py2many/tests/exception/test_exceptions.jl
module test_exceptions
using FromFile: @from
using ParameterTests
using Test

@from "../../exception/__init__.jl" using exception:
    Error, InvalidFormatError, InvalidTargetError
@paramtest "test_error" begin
    @given err ∈ [Error, InvalidFormatError, InvalidTargetError]
    e = err("test")
    @test(e.message == "test")
    @test(String(e) == repr("test"))
end

end

File: libraries/python-reprojector/pyreproj/tests/exception/test_exceptions.py
# -*- coding: utf-8 -*-
import pytest
from pyreproj.exception import Error, InvalidFormatError, InvalidTargetError


@pytest.mark.parametrize("err", [
    Error,
    InvalidFormatError,
    InvalidTargetError
])
def test_error(err):
    e = err('test')
    assert e.message == 'test'
    assert str(e) == repr('test')
